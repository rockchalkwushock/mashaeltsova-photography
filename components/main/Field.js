import PropTypes from 'prop-types'

import { styles } from '../../lib'

const Field = props => {
  if (props.className === 'button') {
    return (
      <div className={props.className}>
        <button type={props.type}>{props.text}</button>
        <style jsx>{`
          .button {
            grid-area: bookingButton;
            grid-column: 1 / 3;
            grid-row: 5 / 6;
          }
          button {
            background-color: ${styles.colors.warm};
            -webkit-box-shadow: 4px 4px 4px rgba(54, 69, 79, 0.7);
            box-shadow: 4px 4px 4px rgba(54, 69, 79, 0.7);
            color: ${styles.colors.white};
            font-family: ${styles.fonts.courgette};
            font-size: 1em;
            padding: 1em;
            text-transform: uppercase;
            width: 100%;
          }
          button:focus,
          button:hover {
            background-color: ${styles.colors.powder};
            color: ${styles.colors.warm};
            font-size: 1.2em;
            -webkit-transition: background-color 0.3s ease-out;
            -moz-transition: background-color 0.3s ease-out;
            transition: background-color 0.3s ease-out;
          }
          @media (min-width: 768px) {
            .button {
              grid-area: bookingButton;
              grid-column: 1 / 3;
              grid-row: 4 / 5;
            }
          }
        `}</style>
      </div>
    )
  }
  return (
    <div className={props.className}>
      <label htmlFor={props.className}>{props.text}</label>
      <input
        name={props.className}
        onChange={props.onChange}
        type={props.type}
        value={props.value}
      />
      <span className="errors">{props.error}</span>
      <style jsx>{`
        label {
          display: block;
        }
        input {
          border: 0;
          border-bottom: 1px solid ${styles.colors.warm};
          color: ${styles.colors.charcoal};
          outline: 0;
          padding: 0 0.5em;
          -webkit-appearance: none;
          width: 100%;
        }
        .errors {
          color: red;
          font-size: 0.75em;
        }
        .firstName {
          grid-area: firstName;
          grid-column: 1 / 2;
          grid-row: 1 / 2;
        }
        .lastName {
          grid-area: lastName;
          grid-column: 2 / 3;
          grid-row: 1 / 2;
        }
        .email {
          grid-area: email;
          grid-column: 1 / 3;
          grid-row: 2 / 3;
        }
        .phone {
          grid-area: phone;
          grid-column: 1 / 3;
          grid-row: 3 / 4;
        }
        .message {
          grid-area: message;
          grid-column: 1 / 3;
          grid-row: 4 / 5;
        }
        @media (min-width: 768px) {
          .email {
            grid-area: email;
            grid-column: 1 / 2;
            grid-row: 2 / 3;
          }
          .phone {
            grid-area: phone;
            grid-column: 2 / 3;
            grid-row: 2 / 3;
          }
          .message {
            grid-area: message;
            grid-column: 1 / 3;
            grid-row: 3 / 4;
          }
        }
      `}</style>
    </div>
  )
}

Field.propTypes = {
  props: PropTypes.shape({
    className: PropTypes.string.isRequired,
    error: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    text: PropTypes.string.isRequired,
    type: PropTypes.string,
    value: PropTypes.string
  })
}

export default Field
