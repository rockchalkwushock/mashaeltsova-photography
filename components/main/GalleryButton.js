import Link from 'next/link'

import { styles } from '../../lib'

export const GalleryLink = () => (
  <a className="gallery-btn">
    Visit Gallery
    <style jsx>{`
      .gallery-btn {
        align-self: center;
        background-color: ${styles.colors.peach};
        border: 0;
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        color: rgba(223, 116, 74, 0.9);
        font-family: ${styles.fonts.courgette};
        height: 2em;
        letter-spacing: 0.25em;
        outline: 0;
        position: absolute;
      }
      .gallery-btn:focus,
      .gallery-btn:hover {
        align-self: center;
        background-color: ${styles.colors.peach};
        border: 0;
        color: ${styles.colors.warm};
        font-size: 1.4em;
        height: 2em;
        outline: 0;
        -webkit-transition: 0.2s ease;
        -moz-transition: 0.2s ease;
        transition: 0.2s ease;
      }
    `}</style>
  </a>
)

const GalleryButton = () => (
  <Link prefetch href="/gallery">
    <GalleryLink />
  </Link>
)

export default GalleryButton
