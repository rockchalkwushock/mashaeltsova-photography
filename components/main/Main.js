import { styles } from '../../lib'

const Main = ({ children }) => (
  <main>
    {children}
    <style jsx>{`
      main {
        background-color: ${styles.colors.powder};
        display: grid;
        grid-gap: 1em;
        grid-template-areas: 'about about about about'
          'services services services services'
          'gallery gallery gallery gallery' 'booking booking booking booking';
        grid-template-columns: repeat(4, 1fr);
        padding: 1em;
      }
    `}</style>
  </main>
)

export default Main
