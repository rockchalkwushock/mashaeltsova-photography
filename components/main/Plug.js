import { P } from '../commons'
import { styles } from '../../lib'

const Plug = () => (
  <div>
    <P className="plug">
      Contact me if you want to relive those precious moments for years to come!
    </P>
    <style jsx>{`
      div {
        background-color: ${styles.colors.peach};
        font-family: ${styles.fonts.courgette};
        padding: 1em;
      }
    `}</style>
  </div>
)

export default Plug
