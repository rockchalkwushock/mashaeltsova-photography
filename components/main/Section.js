import PropTypes from 'prop-types'

import { styles } from '../../lib'

const Section = ({ children, className }) => (
  <section className={className}>
    {children}
    <style jsx>{`
      .about {
        background-color: ${styles.colors.lemon};
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        color: ${styles.colors.charcoal};
        grid-area: about;
        grid-column: 1 / 5;
        padding: 0.75em;
        text-align: left;
      }
      .services {
        background-color: ${styles.colors.lemon};
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        color: ${styles.colors.charcoal};
        display: flex;
        flex-direction: column;
        grid-area: services;
        grid-column: 1 / 5;
        padding: 0.75em;
        text-align: left;
      }
      .gallery {
        background-color: ${styles.colors.peach};
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        display: flex;
        grid-area: gallery;
        grid-column: 1 / 5;
        height: 250px;
        justify-content: center;
      }
      .booking {
        box-shadow: 0 0 20px rgba(54, 69, 79, 0.7);
        grid-area: booking;
        grid-column: 1 / 5;
      }
      @media (min-width: 540px) {
        .gallery {
          height: 360px;
        }
      }
      @media (min-width: 768px) {
        .about {
          grid-column: 1 / 3;
          grid-row: 6 / 8;
        }
        .services {
          grid-column: 3 / 5;
          grid-row: 6 / 8;
        }
        .gallery {
          grid-column: 1 / 5;
          grid-row: 1 / 6;
        }
        .booking {
          display: grid;
          grid-template-columns: 1fr 2fr;
          grid-column: 1 / 5;
          grid-row: 8 / 12;
        }
      }
      @media (min-width: 1024px) {
        .gallery {
          height: 425px;
        }
      }
    `}</style>
  </section>
)

Section.propTypes = {
  className: PropTypes.string.isRequired
}

export default Section
