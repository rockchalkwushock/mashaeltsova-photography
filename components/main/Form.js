import { styles } from '../../lib'

import Field from './Field'

const Form = () => (
  <form>
    <Field className="firstName" text="First Name" type="text" />
    <Field className="lastName" text="Last Name" type="text" />
    <Field className="email" text="Email" type="email" />
    <Field className="phone" text="Phone" type="tel" />
    <Field className="message" text="Short Message" type="text" />
    <Field className="button" text="Book Now!" type="submit" />
    <style jsx>{`
      form {
        background-color: ${styles.colors.white};
        display: grid;
        grid-template-areas: 'firstName firstName' 'lastName lastName'
          'email email' 'phone phone' 'message message'
          'bookingButton bookingButton';
        grid-template-columns: repeat(2, 1fr);
        grid-gap: 1em;
        padding: 1em;
        text-align: left;
      }
    `}</style>
  </form>
)

export default Form
