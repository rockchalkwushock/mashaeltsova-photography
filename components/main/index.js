import CollagePhoto from './CollagePhoto'
import GalleryButton from './GalleryButton'
import Main from './Main'
import Section from './Section'

export { CollagePhoto, GalleryButton, Main, Section }
