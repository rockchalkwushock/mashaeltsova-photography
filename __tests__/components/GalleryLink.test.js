import React from 'react'
import { render } from 'enzyme'

import { GalleryLink } from '../../components/main/GalleryButton'

describe('Component: GalleryLink', () => {
  test('Renders without exploding', () => {
    const tree = render(<GalleryLink />)
    expect(tree).toMatchSnapshot()
  })
})
