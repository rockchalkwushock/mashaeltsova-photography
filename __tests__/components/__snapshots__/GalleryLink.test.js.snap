// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GalleryLink Renders without exploding 1`] = `
<a
  class="jsx-1424217103 gallery-btn"
>
  Visit Gallery
</a>
`;
